providers/netty4/src/main/java/org/asynchttpclient/netty/handler/Processor.java

```diff
@@ -71,11 +75,41 @@ public void channelRead(final ChannelHandlerContext ctx, Object msg) throws Exce
                 ac.call();
                 Channels.setDiscard(channel);
             }
+            ReferenceCountUtil.release(msg);
+
 
         } else if (attribute instanceof NettyResponseFuture) {
             NettyResponseFuture<?> future = (NettyResponseFuture<?>) attribute;
             protocol.handle(channel, future, msg);
 
+        } else if (attribute instanceof StreamedResponsePublisher) {
```

Comment:
@dotta I believe https://github.com/dotta/async-http-client/blob/issue/544-reactive-streams-support/providers/netty4/src/main/java/org/asynchttpclient/netty/handler/Processor.java#L177 is where the retry is initiated, and that code is only executed if the attribute is a `NettyResponseFuture`, so that's where we need to change things.